demo/src/test/java/com/example/test/core/keywork/CustomKeyword.java

```diff
@@ -1,151 +1,94 @@
-            JavascriptExecutor js = (JavascriptExecutor) driver;
-            js.executeScript("arguments[0].scrollIntoView(true);", element);
-            Thread.sleep(3000);
-        } catch (InterruptedException e) {
-            e.printStackTrace();
+            scrollToElemtnIntoView(driver.findElement(By.xpath(locator)));
+            return wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath(locator)));
+        } catch (WebDriverException ex) {
+            throw new WebDriverException("Element not found!");
         }
     }
 
     /**
-     * Check element is display :v
+     * Scroll to element into View
      * 
      * @param element
-     * @return
      */
-    public boolean isDisplay(WebElement element){
-        return waitForElementIsDisplayed(element).isDisplayed();
+    public void scrollToElemtnIntoView(WebElement element) {
+        ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true);", element);
     }
 
-    /**
-     * Wait to element visible
-     * 
-     * @param element to check display and clickable 
-     * @return element to be located
-     */
-    public WebElement waitForElementIsDisplayed(WebElement element){
+    public WebElement waitForElementDisplayed(WebElement element) {
         try {
+            scrollToElemtnIntoView(element);
             return wait.until(ExpectedConditions.elementToBeClickable(element));
-        } catch (WebDriverException e) {
-            throw new WebDriverException("Element is not displayed ");
+        } catch (WebDriverException ex) {
+            throw new WebDriverException("Element not displayed");
         }
+
     }
 
     /**
-     * Check place holder equal given text or not
+     * Wait for Element displayed and input text
      * 
      * @param element
-     * @return
+     * @param text
+     * @throws InterruptedException
      */
-    public boolean checkPlaceHolder(WebElement element, String text){
-        String attribute = waitForElementIsDisplayed(element).getAttribute("placeholder");
-        return (attribute.equals(text));
+    public void sendKeys(WebElement element, String text) throws InterruptedException {
+        try {
+            waitForElementDisplayed(element).sendKeys(text);
+            Thread.sleep(2000);
+        } catch (WebDriverException ex) {
+            throw new WebDriverException("Element not availabe to input text!");
+        }
     }
 
-    /**
-     * Click to element
-     * 
-     * @param element to click
-     */
-    public CustomKeyword click(WebElement element){
-        Actions actions = new Actions(this.driver);
-        actions.moveToElement(element).build().perform();
-        waitForElementIsDisplayed(element).click();
-        return new CustomKeyword(driver);
+    public CustomKeyword clickThenWait(WebElement element) throws InterruptedException {
+        try {
+            waitForElementDisplayed(element);
+            element.click();
+            Thread.sleep(5000);

```

Comment:
should remove thread.sleep and find the way to handle timeout

Reply:
Removed thread sleep and use implicitlyWait